pulp_rpm/app/modelresource.py

```diff
@@ -538,7 +538,6 @@ def set_up_queryset(self):
                     update_record__in=UpdateRecord.objects.filter(pk__in=self.repo_version.content)
                 )
             )
-            .distinct("name", "epoch", "version", "release", "arch")
```

Comment:
Annotation for myself:

>If a rpm exist in multiple errata, then only one rpm among the "update collection packages" will be picked to export.

> For example: oci-umount rpm is associated in multiple errata as shown in the SOURCE Satellite below:
~~~
# select id, digest,ucp.filename, uc.name from rpm_updaterecord r left join rpm_updatecollection uc on uc.update_record_id = r.content_ptr_id left join rpm_updatecollectionpackage ucp on uc.pulp_id = ucp.update_collection_id where ucp.name = 'oci-umount' and ucp.version = '2.3.3' and uc.name = 'rhel-7-for-power-le-extras-rpms__ppc64le_0_default' order by r.id;
       id       |                              digest                              |                   filename                    |                        name                        
----------------+------------------------------------------------------------------+-----------------------------------------------+----------------------------------------------------
 RHBA-2018:0434 | 60dd7201b18e3a27ad706b04a97d9c8edbd88d5301554dd129b6e87af995b99b | oci-umount-2.3.3-3.gite3c9055.el7.ppc64le.rpm | rhel-7-for-power-le-extras-rpms__ppc64le_0_default
 RHEA-2018:1063 | d1fa6f891c2aefb3eb4e443a9254aa96e72a02581f8218b99088766680cf10c6 | oci-umount-2.3.3-3.gite3c9055.el7.ppc64le.rpm | rhel-7-for-power-le-extras-rpms__ppc64le_0_default
 RHEA-2018:3365 | 609a77eec590a3fc55f1d44e36ab25e20d7231c38536d839776dbaba82e9b833 | oci-umount-2.3.3-3.gite3c9055.el7.ppc64le.rpm | rhel-7-for-power-le-extras-rpms__ppc64le_0_default
~~~

> But only one update collection package is imported to the DESTINATION Satellite as shown below:
~~~
# select id, digest,ucp.filename, uc.name from rpm_updaterecord r left join rpm_updatecollection uc on uc.update_record_id = r.content_ptr_id left join rpm_updatecollectionpackage ucp on uc.pulp_id = ucp.update_collection_id where ucp.name = 'oci-umount' and ucp.version = '2.3.3' and uc.name = 'rhel-7-for-power-le-extras-rpms__ppc64le_0_default' order by r.id;
       id       |                              digest                              |                   filename                    |                        name                        
----------------+------------------------------------------------------------------+-----------------------------------------------+----------------------------------------------------
 RHBA-2018:0434 | 60dd7201b18e3a27ad706b04a97d9c8edbd88d5301554dd129b6e87af995b99b | oci-umount-2.3.3-3.gite3c9055.el7.ppc64le.rpm | rhel-7-for-power-le-extras-rpms__ppc64le_0_default
~~~

My question is, why was this `.distinct()` here and will we suffer consequences for removing it?  The origin was https://github.com/pulp/pulp_rpm/commit/36afce0e85977a6f9eb300ad563bf286919c78a5 which was intended to resolve https://github.com/pulp/pulp_rpm/issues/2648

Reply:
It's getting late and my brain is fried so this could be misinterpretation.  Is it possible that originally the `.distinct()` was intended to deduplicate amongst update collection packages within one update collection, but the end result was in fact deduplicating across all update collections and butchering them in the process?

Reply:
However, I think that commit updates both the import and output side, so even if we drop the `.distinct()` entirely we might not see any negative impact.  If that is the case then this patch as-is would not break anything if applied as-is, which may be helpful if we're in a really time-sensitive spot.

Reply:
OK, so, for posterity, let me braindump a little:

- UpdateRecords have a "digest", which takes into account **everything** about the record - content, ref-lists, and package-collections
- UpdateCollections are unique-by-name for a given UpdateRecord
- UpdateCollectionPackages (UCP) can exist in multiple UpdateCollections within a given UpdateRecord
- at export-time we map a specific UpdateCollectionPackageResource (UCPR) to the specific UpdateCollection it belongs in, by collection-name|updaterecord-digest

So - the only thing the "distinct" that Hao discovered could be removed was doing, was (incorrectly) removing UCPR records that were shared between UCPs of the same UpdateRecord.

This was originally done to deal with an oddness around updateinfo.xmls in the Alma repositories. With this PR as-is, I have been able to export/import the Alma base-repo (which used to fail consistently before #2648 was addressed.

We have responded to a number of "rpm ecosystem data is Deeply Weird" issues, it's possible the particular Alma oddness was resolved "higher up" in the sync-process **after** #2648 was addressed. Or Alma changed/fixed their updateinfo.xml generation. Or the distinct() was never *really* needed (my current theory). 

I am continuing to experiment, but so far, this patch is looking good.